index.rst

```diff
@@ -1,61 +1,338 @@
+    This will be used as the username for Science Platform services and must not contain a ``@`` character (or any other character that causes issues for Science Platform services).
+    The account name must not match the name of any existing account.
+
+IDM-0011
+    Account names may not be reused except via (rare) administrative intervention to remove accounts that were erroneously created.
+
+IDM-0012
+    Users must be able to change their account name.
+    This change must propagate to all Science Platform services.
+    Science Platform services should therefore use a unique identifier rather than the account name (such as the numeric UID provided as part of the account metadata) wherever possible, and if not possible, must explicitly allow for account renaming.
+
+Token authentication
+--------------------
+
+IDM-0100
+    Users may generate access tokens to use with API calls to the Science Platform.
+    Access tokens must have an associated name chosen by the user.
+    Users must be able to see a list of all their current access tokens (the names only, not the values) and revoke any access token.
+
+IDM-0101
+    User access tokens must not grant access to any administrative Science Platform function or permit changes to a user's account metadata or authentication information.
+    Specifically, an access token cannot be used to attach a new federated identity to an account, revoke a federated identity from an account, or change a user's email address or group membership.
+
+IDM-0102
+    Users may set an expiration time on user-generated access tokens.
+    By default, user-generated access tokens do not expire, although their permisisons are tied to the user's own permissions and thus they may become unusuable if the account is frozen or deleted or its access permissions change.
+
+IDM-0103
+    Internal Science Platform components may also generate temporary access tokens to facilitate multi-layer services.
+    These tokens must also be shown on the user's token list and the user must be able to revoke them, but they should be visually separated from user-created access tokens.
+    Internal temporary access tokens must expire in a reasonable length of time, such as shortly after the expected maximum duration of the operation for which they were intended.
+
+IDM-0104
+    Tokens should be scoped to restrict their power.
+    However, the number of scopes should not be so large as to be overwhelming.
+    A user should be able to easily choose the necessary scope of a token for common token-based workflows.
+    User-visible scopes should be limited to at most a few dozen, preferrably fewer.
+
+IDM-0105
+    Tokens must not contain a frozen representation of group membership or permissions.
+    Updates to the group membership of a user's account should also apply to all tokens issued for that user, provided that the scope of the token allows access.
+    See `SQR-039 <https://sqr-039.lsst.io/>`__ for more discussion.
```

Comment:
It might be clearer to say here that groups won't be in the tokens, but you use the tokens to determine the groups?  That's my reading of section 4.3 of sqr-39.  I guess that's an implementation detail, but an important one.

I think you're trying to say that without saying an implementation here, but it might be helpful to say "should also apply to all CURRENT tokens issued"  - the idea that existing tokens don't need to be reissued for the groups to be valid?

Reply:
I added a note to make that a bit more explicit that any services that need to know group membership information must present that token to the identity management system and ask, and the answer may be change over the lifetime of the token.